### PR TITLE
Add RothC initialisation

### PR DESCRIPTION
**Pull request by FrankvRaffe**
Added several options to initialise C pool distribution in the RothC calculation

## Summary by CodeRabbit

* **New Features**
  * Named initialization methods (multiple spin‑up options) that return explicit carbon pools; scenario input accepts cf_yield; tillage-based rate modifier, irrigation handling and a visualize option added.

* **Bug Fixes**
  * Stricter input validation (fractions, units, output granularity) and clearer error messages; safer initialization and numeric-edge-case checks.

* **Documentation**
  * Expanded help pages and NEWS entry describing new initialization options and defaults.

* **Tests**
  * Tests extended to cover all initialization paths, tillage/irrigation effects and new parameters.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by FrankvRaffe** [comment]
@coderabbitai review

File: R/rc_sim.R [hunk 1]
```diff
@@ -42,7 +41,7 @@
 #'
 #' rothc_parms: parameters to adapt calculations (optional)
 #' May include the following columns:
-#' * initialize (boolean) scenario to initialize the carbon pools. Options TRUE or FALSE, default is FALSE
+#' * type (character) scenario to initialize the carbon pools. options 'spinup_analytical_bodemcoolstof','spinup_analytical_heuvelink', 'spinup_simulation', 'none', default is 'spinup_analytical_bodemcoolstof'
```
Comment (KeesDoolNMI) [inline R/rc_sim.R]: I think `initialization_method` is a better name, `type` may refer to anything

File: R/rc_sim.R [hunk 2]
```diff
@@ -161,78 +159,45 @@ rc_sim <- function(soil_properties,
...
-    # Set the intial C pools (kg C / ha)
-    dt.soc[,CIOM0 := ciom.ini * 1000]
-    dt.soc[,CDPM0 := cdpm.ini * 1000]
-    dt.soc[,CRPM0 := crpm.ini * 1000]
-    dt.soc[,CBIO0 := cbio.ini * 1000]
-    dt.soc[,CHUM0 := chum.ini * 1000]
-  } else {
+
+  # initialize the RothC pools (kg C / ha)
+  if(type != 'none'){
```
Comment (KeesDoolNMI) [inline R/rc_sim.R]: What happens when `type == 'none'` ?

Comment (FrankvRaffe) [reply R/rc_sim.R]: No initialization happens and the supplied (or default) pool distribution is used to calculate initial pool sizes. I have added a short line of documentation to clarify

File: R/rc_initialise.R [hunk 3]
```diff
@@ -1,38 +1,361 @@
...
+#' @param amendment (data.table) A table with the following column names: year, month, cin_tot, cin_hum, cin_dpm, cin_rpm and the fraction eoc over p (fr_eoc_p). Month is optional.
+#' @param dt.soc (data.table) Data table containing information on soil properties. See details for information.
+#' @param dt.time (data.table) Combination of months and years of the entire simulation period. 
+#' @param rothc.parms (list) List with relevant RothC parameters. See details for more information
+#' @param rothc.event (data.table) List with events of C inputs. See details for more information
+#' @param start_date (character, formatted YYYY-MM-DD) start date of the simulation, required if type is set to spinup_simulation
+#' @param soil_properties (list) list of relevant soil properties, required if type is set to spinup_simulation
+#' @param dt.weather (data.table) average weather conditions for the location of interested, recommended if type is set to spinup_simulation
+#' @param type (character) options for spin-up (spinup_simulation,spinup_analytical_bodemcoolstof, spinup_analytical_heuvelink)
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_initialise.R]: Agreed! Altered the name in the new commits

File: R/rc_input_scenario.R
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_input_scenario.R]: Good idea!

File: tests/testthat/test-rc_helpers.R [hunk 4]
```diff
@@ -285,7 +285,7 @@ test_that("rc_update_parms accepts and validates poutput", {
   expect_equal(result$poutput, "year")
   
   # Test invalid poutput
-  expect_error(rc_update_parms(list(poutput = "invalid"), crops = crops), "additional elements")
+  expect_error(rc_update_parms(list(poutput = "invalid"), crops = crops), "has additional elements")
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply tests/testthat/test-rc_helpers.R]: @coderabbitai I had already changed the expectation to "element of set". I am not sure what code you are looking at.

File: R/rc_initialise.R [hunk 5]
```diff
@@ -1,38 +1,363 @@
...
+    
+    # Set newly required inputs
+  
+    # Run initialization run for 150 years
+    this.result <- rc_sim(rothc_rotation = crop_extend,
+                          rothc_amendment = amendment_extend,
+                          soil_properties = soil_properties,
+                          weather = dt.weather,
+                          rothc_parms = parms)
+
```
Comment (gerardhros) [inline R/rc_initialise.R]: seems ok, but by setting rothc_parms = parms you are using the default k1 to k4 values and abc. 
and do you need a check that soil_properties had only one option (or is that checked within rc_sim?)

File: R/rc_initialise.R [hunk 6]
```diff
@@ -1,38 +1,363 @@
...
   
-  # Prepare input for scenario Business As Usual
-  scen.inp <- rc_input_scenario(B_LU_BRP = B_LU_BRP, scen = 'BAU')
-  rotation <- scen.inp$rotation
-  amendment <- scen.inp$amendment
   
-  # Set model parameters
-  parms <- list(simyears = 150,unit = 'psomperfraction', initialize = TRUE)
+    fractions <- this.result.fin[,.(fr_IOM = CIOM / A_SOM_LOI,
+                                    fr_DPM = CDPM / A_SOM_LOI,
```
Comment (gerardhros) [inline R/rc_initialise.R]: check: is fr_DPM the fraction of A_SOM_LOI or should it be the fraciton from A_COM_LOI - CIOM?

File: R/rc_initialise.R [hunk 7]
```diff
@@ -1,38 +1,363 @@
...
+    if (is.null(crops) || nrow(crops) == 0) {
+      c_input_crop <- 0
+    }else{
+    c_input_crop <- crops[,sum(B_C_OF_INPUT)/isimyears]
+    }
+    
+    if (is.null(amendment) || nrow(amendment) == 0) {
+       c_input_man <- 0
+    } else if (!is.null(amendment$B_C_OF_INPUT) && any(!is.na(amendment$B_C_OF_INPUT))) {
+        c_input_man <- amendment[, sum(B_C_OF_INPUT)/isimyears]
```
Comment (gerardhros) [inline R/rc_initialise.R]: is this now a difficult way to check that "B_C_OF_INPUT" %in% colnames(amendment)?
why adding the second check, since you are also not doing that for P_DOSE below (you add na.rm=T).
so, it can be c_input_man <- amendment[, sum(B_C_OF_INPUT, na.rm=TRUE)/isimyears]

File: R/rc_initialise.R [hunk 8]
```diff
@@ -1,38 +1,363 @@
...
+    }
   
-  # take last two rotations
-  this.result.fin <- this.result[year > max(year)-2*nrow(rotation),lapply(.SD,mean)]
+
+    # calculate C input ratio of crop and amendments
+    c_input_tot <- c_input_crop+c_input_man
+    
+    if(c_input_tot <= 0 || !is.finite(c_input_tot)){
+      # if there are no C inputs, continue calculation based on crops only to prevent NaNs
```
Comment (gerardhros) [inline R/rc_initialise.R]: does this option happen? probably only when c_input_tot = 0?

File: R/rc_initialise.R [hunk 9]
```diff
@@ -1,38 +1,363 @@
...
+      CR_proportion <- 1
+      M_proportion <- 0
+    }else{
+    CR_proportion <- c_input_crop / (c_input_tot)
+    M_proportion <- c_input_man / (c_input_tot)
+    }
+    
+    # estimate DPM-RPM ratio of the inputs from crop and amendments
+    if (!is.null(crops) && nrow(crops) > 0) {
```
Comment (gerardhros) [inline R/rc_initialise.R]: i have the impression from some repetition here.
earlier you already did a check whether crops is there, same for amendment. 
if so, then within that if statement you can in principle also apply these calculations isn't?
that minimises the number of checks.

File: R/rc_initialise.R [hunk 10]
```diff
@@ -1,38 +1,363 @@
...
+    # Define ratio CO2 / (BIO+HUM)
+    x <- 1.67 * (1.85 + 1.60 * exp(-0.0786 * dt.soc$A_CLAY_MI))
+    
+    # transfer coefficients upon C degradation (B is BIO, H = HUM)
+    B = 0.46/(x + 1)
+    H = 0.54/(x + 1)
+    
+    # rates of the five pools, DPM, RPM, BIO, HUM and IOM 
+    ks <- c(k1, k2, k3, k4, 0)
+
```
Comment (gerardhros) [inline R/rc_initialise.R]: would you allow them to be different than the default RothC rates?

File: R/rc_initialise.R (hunk 6 above)
Comment (gerardhros) [reply R/rc_initialise.R]: ok, is see that you later do the same for heuvelink method. it is consistent. Good to check with RothC documentation. if similar, than its fine.

File: R/rc_initialise.R [hunk 11]
```diff
@@ -1,38 +1,363 @@
 #' Function for initializing the RothC model
 #'
```
Comment (gerardhros) [inline R/rc_initialise.R]: is this function applicable for multiple fields only? it seems only applicable per field, and then its good to mention.

File: R/rc_initialise.R [hunk 12]
```diff
@@ -1,38 +1,363 @@
...
+    
+    # recalculate total organic carbon from kg to ton C / ha
+    dt.soc[,toc := toc/1000]
+    
+    # time correction (is 12 in documentation Chantals' study, not clear why, probably due to fixed time step calculation)
+    timecor = 1
+
+    # CDPM pool (ton C / ha)
+    cdpm.ini <- rothc.event[var == 'CDPM',list(time,value)]
+    if(nrow(cdpm.ini) >0){
```
Comment (gerardhros) [inline R/rc_initialise.R]: check question: value has units kg total organic carbon/ha?

File: R/rc_sim.R [hunk 13]
```diff
@@ -185,77 +181,46 @@ rc_sim <- function(soil_properties,
...
+                                        soil_properties = soil_properties,
+                                        rothc.event = rothc.event,
+                                        start_date = start_date,
+                                        dt.weather = weather,
+                                        rothc.parms = rothc.parms,
+                                        initialization_method = initialization_method))
     
-  }
   
+  }
```
Comment (gerardhros) [inline R/rc_sim.R]: there is no else command? should you have c_fractions <- rothc_parms$c_fractions?

File: R/rc_initialise.R (hunk 10 above)
Comment (FrankvRaffe) [reply R/rc_initialise.R]: These values can indeed be either user input or default rates (if non supplied)

File: R/rc_sim.R [hunk 14]
```diff
@@ -76,7 +73,6 @@
 rc_sim <- function(soil_properties,
                    A_DEPTH = 0.3,
                    B_DEPTH = 0.3,
-                   M_TILLAGE_SYSTEM = 'CT',
                    rothc_rotation = NULL,
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_sim.R]: Correct, I could not find this in the original RothC documentation and have thus omitted it to follow the concept of "simple and straightfoward" as much as possible. We can discuss whether we can still add it as an option to get more realistic results from certain management methods.

**Review by KeesDoolNMI — APPROVED** [review]
Great work!

File: R/rc_sim.R (hunk 14 above)
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_sim.R]: I will do a short literature research to see what I can find, and provide some documentation (perhaps something to add in the vignettes?). I do feel, as states above, that this implementation should be outside of the rotsee package and that based on practices users can provide updated k values. But I am open to be persuaded to a different approach.

File: R/rc_initialise.R (hunk 8 above)
Comment (gerardhros) [reply R/rc_initialise.R]: okido. i see that for first but it can never be infinite isn't? the value is either zero or its a value. but ok to leave it in.

File: R/rc_sim.R (hunk 13 above)
Comment (gerardhros) [reply R/rc_sim.R]: okido

File: R/rc_sim.R (hunk 14 above)
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_sim.R]: Extra literature is welcome! And I see, implementing as an rmf makes sense, I'll add a temporary rmf for now then.

File: tests/testthat/test-rc_initialise.R [hunk 15]
```diff
@@ -0,0 +1,539 @@
...
+    expect_named(result, c("fr_IOM", "fr_DPM", "fr_RPM", "fr_BIO"))
+    
+    # Check values are valid fractions
+    expect_true(all(result >= 0))
+    expect_true(all(result <= 1))
+    expect_true(all(is.finite(result)))
+    
+    # Check sum is less than or equal to 1
+    expect_lte(sum(result), 1)
+  }
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply tests/testthat/test-rc_initialise.R]: @coderabbitai changed!

File: R/rc_sim.R [hunk 16]
```diff
@@ -185,77 +186,53 @@ rc_sim <- function(soil_properties,
...
-    # Calculate carbon pools based on provided or default distribution (kg C / ha)
-    dt.soc[,CIOM0 := c_fractions$fr_IOM * ((toc*0.001)^1.139) * 1000]
+    # calculate initial pool sizes based on supplied fraction distribution (kg C / ha)
+    dt.soc[,CIOM0 := c_fractions$fr_IOM * toc^1.139]
     dt.soc[,CDPM0 := c_fractions$fr_DPM * (toc-CIOM0)]
     dt.soc[,CRPM0 := c_fractions$fr_RPM * (toc-CIOM0)]
     dt.soc[,CBIO0 := c_fractions$fr_BIO * (toc-CIOM0)]
-    dt.soc[,CHUM0 := toc-CIOM0-CDPM0-CRPM0-CBIO0]
-    
+    dt.soc[,CHUM0 := pmax(0, toc-CIOM0-CDPM0-CRPM0-CBIO0)]
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_sim.R]: @coderabbitai fr_IOM is distinctly different from the other fractions. I have removed it from the checkmate in rc_update_parms. The other fractions should be below 1 as they are the fractions of the reactive C pool (so without IOM)

File: R/rc_initialise.R [hunk 17]
```diff
@@ -1,43 +1,376 @@
...
+    }
+  
+  
+  # Define rothc parameters
+  k1 <- rothc.parms$k1
+  k2 <- rothc.parms$k2
+  k3 <- rothc.parms$k3
+  k4 <- rothc.parms$k4
+  
+  abc <- rothc.parms$abc
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply R/rc_initialise.R]: @coderabbitai Updated!
